src/Microsoft.DotNet.Interactive/Kernel.cs

```diff
@@ -215,6 +218,19 @@ public FrontendEnvironment FrontendEnvironment
 
         public void AddDirective(Command command) => SubmissionParser.AddDirective(command);
 
+        public virtual void RegisterCommandHandler<TCommand>(string commandType, Func<TCommand, KernelInvocationContext, Task> handler)
```

Comment:
Does this need to be `virtual`?

Reply:
No. At the time, I was thinking that different kernels might want to handle this differently, but it was a vague thought. Since I don't have a clear idea of what an `override` of this might actually do, I've pushed a commit making it non-`virtual`.